src/sync/sync_session.cpp

```diff
@@ -636,8 +636,15 @@ void SyncSession::handle_error(SyncError error)
             }
             break;
         case NextStateAfterError::inactive: {
-            std::unique_lock<std::mutex> lock(m_state_mutex);
-            advance_state(lock, State::inactive);
+            if (error.is_client_reset_requested()) {
```

Comment:
Why not use this error instead of constructing a new one with `make_error_code`?

Reply:
I agree, I think you should be able to use `error.error_code` instead of making a `diverging_histories`